### PR TITLE
Fix snapshot build, upgrade to OpenSearch 1.1.

### LINKED ISSUE #48: [BUG] Snapshot build is broken
**What is the bug?**

Fix and default CI to building a snapshot build. 
Use OpenSearch 1.1.

**What is the expected behavior?**

Build output produces 1.1.0-SNAPSHOT package and JARs in maven.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**

- Upgrades version to 1.1, uses OpenSearch version.
- Runs snapshot builds by default. 
- Renames output artifact to include version.
- Disables slow tests (see #57)
- Removes ODFE from docker container.

Closes #48, #56, #40.

There's a circular dependency described in https://github.com/opensearch-project/performance-analyzer/issues/50, so https://github.com/opensearch-project/performance-analyzer/pull/53 is needed. Once this is merged I'll switch references to my fork that contains this code.

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer-rca/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
I think I need to fix performance-analyzer first ...

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -49,34 +53,36 @@ jobs:
...
       uses: actions/checkout@v2
       with:
-        repository: 'opensearch-project/OpenSearch'
-        path: OpenSearch
-        ref: '1.0'
-    - name: Build OpenSearch
-      working-directory: ./OpenSearch
-      run: ./gradlew publishToMavenLocal -Dbuild.snapshot=false
+        repository: dblock/performance-analyzer
+        ref: fix-snapshot-build
```
Comment (sruti1312) [inline .github/workflows/gradle.yml:63]: We might need to change this before the merge

**Comment by sruti1312** [comment]
If we are adding version number to the performance-analyzer-rca artifact name then we need to update them here as well: https://github.com/opensearch-project/performance-analyzer-rca/blob/main/docker/Dockerfile#L61

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (dblock) [reply .github/workflows/gradle.yml:63]: There's a circular dependency between these two projects, so we cannot do that without breaking everything. I will update to use main branches after both are in.

**Comment by dblock** [comment]
@sruti1312 Could you please help me with this? I can't get a successful build, tests are timing out and/or are flaking. I opened #57. What should I do with this?

**Comment by dblock** [comment]
I disabled slow integration tests (and re-enabled two tests that aren't failing anymore) in https://github.com/opensearch-project/performance-analyzer-rca/pull/55/commits/b6e510cc683d6e222ac6be93302603323bc960a8, hope we can merge this and #57 can be picked up by the plugin team? @sruti1312

**Review by sruti1312 — APPROVED** [review]

**Pull request merged**